[2023-10-27 18:11:10,378: INFO: main: welcome to our custom log!]
[2023-10-29 11:39:38,864: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-10-29 11:39:38,865: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-10-29 11:39:38,866: INFO: common: yaml file: params.yaml loaded successfully]
[2023-10-29 11:39:38,866: INFO: common: created directory: artifacts]
[2023-10-29 11:39:38,866: INFO: common: created directory: artifacts/data_ingestion]
[2023-10-29 11:39:38,866: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-10-29 11:39:56,821: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-10-29 11:39:57,139: INFO: main: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-16 22:30:54,377: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-16 22:30:54,381: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-16 22:30:54,384: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-16 22:30:54,385: INFO: common: created directory: artifacts]
[2023-11-16 22:30:54,385: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-16 22:30:54,386: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-16 22:31:00,566: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-16 22:31:01,043: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-16 22:31:01,043: INFO: main: *******************]
[2023-11-16 22:31:01,044: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-16 22:31:01,045: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-16 22:31:01,045: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-16 22:31:01,046: INFO: common: created directory: artifacts]
[2023-11-16 22:31:01,046: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-16 22:31:01,375: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-16 22:31:01,663: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-17 18:38:15,551: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 18:38:15,553: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 18:38:15,554: INFO: common: created directory: artifacts]
[2023-11-17 18:39:49,281: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 18:39:49,282: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 18:39:49,283: INFO: common: created directory: artifacts]
[2023-11-17 19:42:52,919: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 19:42:52,922: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 19:42:52,923: INFO: common: created directory: artifacts]
[2023-11-17 19:42:59,099: INFO: common: json file saved at scores.json]
[2023-11-17 19:52:07,287: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 19:52:07,289: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 19:52:07,289: INFO: common: created directory: artifacts]
[2023-11-17 19:52:11,812: INFO: common: json file saved at scores.json]
[2023-11-17 19:52:43,766: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 19:52:43,767: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 19:52:43,768: INFO: common: created directory: artifacts]
[2023-11-17 19:52:48,290: INFO: common: json file saved at scores.json]
[2023-11-17 19:57:09,338: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 19:57:09,339: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 19:57:09,340: INFO: common: created directory: artifacts]
[2023-11-17 19:57:13,830: INFO: common: json file saved at scores.json]
[2023-11-17 19:57:23,697: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-17 19:57:24,095: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmpy6se1ehx\model\data\model\assets]
[2023-11-17 21:37:14,799: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-17 21:37:14,800: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 21:37:14,802: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 21:37:14,802: INFO: common: created directory: artifacts]
[2023-11-17 21:37:14,802: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-17 21:37:14,802: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-17 21:37:20,407: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-17 21:37:20,700: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-17 21:37:20,700: INFO: main: *******************]
[2023-11-17 21:37:20,700: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-17 21:37:20,701: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 21:37:20,702: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 21:37:20,702: INFO: common: created directory: artifacts]
[2023-11-17 21:37:20,702: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-17 21:37:20,931: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-17 21:37:21,060: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-17 21:37:21,060: INFO: main: *******************]
[2023-11-17 21:37:21,060: INFO: main: >>>> stage: Training started <<<<]
[2023-11-17 21:37:21,062: ERROR: main: name 'ModelTrainingPipeline' is not defined]
Traceback (most recent call last):
  File "main.py", line 33, in <module>
    obj = ModelTrainingPipeline()
NameError: name 'ModelTrainingPipeline' is not defined
[2023-11-17 21:38:32,279: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-17 21:38:32,281: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 21:38:32,283: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 21:38:32,284: INFO: common: created directory: artifacts]
[2023-11-17 21:38:32,284: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-17 21:38:32,284: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-17 21:38:37,221: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-17 21:38:37,504: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-17 21:38:37,504: INFO: main: *******************]
[2023-11-17 21:38:37,505: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-17 21:38:37,506: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 21:38:37,507: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 21:38:37,507: INFO: common: created directory: artifacts]
[2023-11-17 21:38:37,507: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-17 21:38:37,732: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-17 21:38:37,859: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-17 21:38:37,859: INFO: main: *******************]
[2023-11-17 21:38:37,859: INFO: main: >>>> stage: Training started <<<<]
[2023-11-17 21:38:37,860: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 21:38:37,861: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 21:38:37,862: INFO: common: created directory: artifacts]
[2023-11-17 21:38:37,862: INFO: common: created directory: artifacts/training]
[2023-11-17 21:38:37,863: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 22:00:09,134: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-17 22:00:09,136: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 22:00:09,137: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 22:00:09,138: INFO: common: created directory: artifacts]
[2023-11-17 22:00:09,138: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-17 22:00:09,138: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-17 22:00:15,231: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-17 22:00:15,536: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-17 22:00:15,537: INFO: main: *******************]
[2023-11-17 22:00:15,537: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-17 22:00:15,538: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 22:00:15,539: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 22:00:15,540: INFO: common: created directory: artifacts]
[2023-11-17 22:00:15,540: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-17 22:00:15,818: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-17 22:00:15,936: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-17 22:00:15,936: INFO: main: *******************]
[2023-11-17 22:00:15,936: INFO: main: >>>> stage: Training started <<<<]
[2023-11-17 22:00:15,937: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 22:00:15,938: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 22:00:15,939: INFO: common: created directory: artifacts]
[2023-11-17 22:00:15,939: INFO: common: created directory: artifacts/training]
[2023-11-17 22:00:15,940: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 22:01:20,786: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-17 22:01:20,786: INFO: main: *******************]
[2023-11-17 22:01:20,786: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-17 22:01:20,788: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-17 22:01:20,789: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-17 22:01:20,789: INFO: common: created directory: artifacts]
[2023-11-17 22:01:25,214: INFO: common: json file saved at scores.json]
[2023-11-17 22:01:26,048: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-17 22:01:26,340: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmp2lcqkj2h\model\data\model\assets]
[2023-11-17 22:01:30,043: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\a500482b651443d88fb8de541a9ca253\artifacts\model\data]
[2023-11-17 22:01:30,044: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\a500482b651443d88fb8de541a9ca253\artifacts\model\data\model]
[2023-11-17 22:01:30,044: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\a500482b651443d88fb8de541a9ca253\artifacts\model\data\model\assets]
[2023-11-17 22:01:30,046: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\a500482b651443d88fb8de541a9ca253\artifacts\model\data\model\variables]
[2023-11-17 22:01:30,088: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-18 01:29:46,448: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-18 01:29:46,450: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:29:46,452: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:29:46,452: INFO: common: created directory: artifacts]
[2023-11-18 01:29:46,453: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-18 01:29:46,453: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 01:29:52,394: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 01:29:52,686: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-18 01:29:52,686: INFO: main: *******************]
[2023-11-18 01:29:52,686: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-18 01:29:52,687: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:29:52,689: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:29:52,689: INFO: common: created directory: artifacts]
[2023-11-18 01:29:52,689: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-18 01:29:52,924: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-18 01:29:53,084: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-18 01:29:53,084: INFO: main: *******************]
[2023-11-18 01:29:53,084: INFO: main: >>>> stage: Training started <<<<]
[2023-11-18 01:29:53,085: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:29:53,087: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:29:53,087: INFO: common: created directory: artifacts]
[2023-11-18 01:29:53,087: INFO: common: created directory: artifacts/training]
[2023-11-18 01:29:53,088: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:30:06,672: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-18 01:30:06,673: INFO: main: *******************]
[2023-11-18 01:30:06,673: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-18 01:30:06,674: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:30:06,675: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:30:06,675: INFO: common: created directory: artifacts]
[2023-11-18 01:30:11,067: INFO: common: json file saved at scores.json]
[2023-11-18 01:30:11,782: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-18 01:30:12,092: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmp6oeg0ttp\model\data\model\assets]
[2023-11-18 01:30:15,369: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\158a719829d447578a3c8996d84b1692\artifacts\model\data]
[2023-11-18 01:30:15,370: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\158a719829d447578a3c8996d84b1692\artifacts\model\data\model]
[2023-11-18 01:30:15,370: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\158a719829d447578a3c8996d84b1692\artifacts\model\data\model\assets]
[2023-11-18 01:30:15,372: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\158a719829d447578a3c8996d84b1692\artifacts\model\data\model\variables]
[2023-11-18 01:30:15,415: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-18 01:34:39,972: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-18 01:34:39,973: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:34:39,974: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:34:39,975: INFO: common: created directory: artifacts]
[2023-11-18 01:34:39,975: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-18 01:34:39,975: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 01:34:45,982: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 01:34:46,277: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-18 01:34:46,277: INFO: main: *******************]
[2023-11-18 01:34:46,277: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-18 01:34:46,279: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:34:46,280: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:34:46,280: INFO: common: created directory: artifacts]
[2023-11-18 01:34:46,280: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-18 01:34:46,517: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-18 01:34:46,644: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-18 01:34:46,644: INFO: main: *******************]
[2023-11-18 01:34:46,645: INFO: main: >>>> stage: Training started <<<<]
[2023-11-18 01:34:46,646: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:34:46,647: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:34:46,647: INFO: common: created directory: artifacts]
[2023-11-18 01:34:46,647: INFO: common: created directory: artifacts/training]
[2023-11-18 01:34:46,648: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:35:00,438: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-18 01:35:00,439: INFO: main: *******************]
[2023-11-18 01:35:00,439: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-18 01:35:00,440: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:35:00,441: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:35:00,442: INFO: common: created directory: artifacts]
[2023-11-18 01:35:04,857: INFO: common: json file saved at scores.json]
[2023-11-18 01:35:05,586: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-18 01:35:05,899: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmpr89xrvqs\model\data\model\assets]
[2023-11-18 01:35:09,291: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\56b88b5048b647b0ab66b64cc5f3b11c\artifacts\model\data]
[2023-11-18 01:35:09,293: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\56b88b5048b647b0ab66b64cc5f3b11c\artifacts\model\data\model]
[2023-11-18 01:35:09,293: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\56b88b5048b647b0ab66b64cc5f3b11c\artifacts\model\data\model\assets]
[2023-11-18 01:35:09,295: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\56b88b5048b647b0ab66b64cc5f3b11c\artifacts\model\data\model\variables]
[2023-11-18 01:35:09,338: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-18 01:46:33,340: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-18 01:46:33,341: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:46:33,343: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:46:33,343: INFO: common: created directory: artifacts]
[2023-11-18 01:46:33,343: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-18 01:46:33,344: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 01:46:39,682: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 01:46:39,974: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-18 01:46:39,974: INFO: main: *******************]
[2023-11-18 01:46:39,974: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-18 01:46:39,975: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:46:39,976: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:46:39,977: INFO: common: created directory: artifacts]
[2023-11-18 01:46:39,977: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-18 01:46:40,203: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-18 01:46:40,329: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-18 01:46:40,330: INFO: main: *******************]
[2023-11-18 01:46:40,330: INFO: main: >>>> stage: Training started <<<<]
[2023-11-18 01:46:40,331: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:46:40,332: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:46:40,332: INFO: common: created directory: artifacts]
[2023-11-18 01:46:40,333: INFO: common: created directory: artifacts/training]
[2023-11-18 01:46:40,334: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:46:54,068: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-18 01:46:54,069: INFO: main: *******************]
[2023-11-18 01:46:54,069: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-18 01:46:54,070: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 01:46:54,071: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 01:46:54,071: INFO: common: created directory: artifacts]
[2023-11-18 01:46:58,501: INFO: common: json file saved at scores.json]
[2023-11-18 01:46:58,502: INFO: common: json file saved at scores.json]
[2023-11-18 01:46:59,234: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-18 01:46:59,559: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmp2p6ni1dr\model\data\model\assets]
[2023-11-18 01:47:02,827: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\03d82ba4725a4f64af3f38a9316b4d40\artifacts\model\data]
[2023-11-18 01:47:02,828: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\03d82ba4725a4f64af3f38a9316b4d40\artifacts\model\data\model]
[2023-11-18 01:47:02,829: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\03d82ba4725a4f64af3f38a9316b4d40\artifacts\model\data\model\assets]
[2023-11-18 01:47:02,830: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\03d82ba4725a4f64af3f38a9316b4d40\artifacts\model\data\model\variables]
[2023-11-18 01:47:02,874: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-18 02:04:01,345: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-18 02:04:01,346: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:04:01,348: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:04:01,348: INFO: common: created directory: artifacts]
[2023-11-18 02:04:01,348: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-18 02:04:01,349: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 02:04:08,308: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 02:04:08,603: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-18 02:04:08,604: INFO: main: *******************]
[2023-11-18 02:04:08,604: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-18 02:04:08,605: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:04:08,606: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:04:08,607: INFO: common: created directory: artifacts]
[2023-11-18 02:04:08,607: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-18 02:04:08,847: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-18 02:04:08,978: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-18 02:04:08,978: INFO: main: *******************]
[2023-11-18 02:04:08,978: INFO: main: >>>> stage: Training started <<<<]
[2023-11-18 02:04:08,979: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:04:08,980: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:04:08,981: INFO: common: created directory: artifacts]
[2023-11-18 02:04:08,981: INFO: common: created directory: artifacts/training]
[2023-11-18 02:04:08,982: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:04:22,740: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-18 02:04:22,741: INFO: main: *******************]
[2023-11-18 02:04:22,741: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-18 02:04:22,742: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:04:22,743: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:04:22,743: INFO: common: created directory: artifacts]
[2023-11-18 02:04:33,650: INFO: common: json file saved at scores.json]
[2023-11-18 02:04:33,651: INFO: common: json file saved at scores.json]
[2023-11-18 02:04:34,374: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-18 02:04:34,662: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmpq9fy9v0n\model\data\model\assets]
[2023-11-18 02:04:37,828: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\9cbb7aed6b784b318ae49f77a38469df\artifacts\model\data]
[2023-11-18 02:04:37,828: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\9cbb7aed6b784b318ae49f77a38469df\artifacts\model\data\model]
[2023-11-18 02:04:37,829: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\9cbb7aed6b784b318ae49f77a38469df\artifacts\model\data\model\assets]
[2023-11-18 02:04:37,831: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\9cbb7aed6b784b318ae49f77a38469df\artifacts\model\data\model\variables]
[2023-11-18 02:04:37,876: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-18 02:37:21,525: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-18 02:37:21,526: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:37:21,528: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:37:21,529: INFO: common: created directory: artifacts]
[2023-11-18 02:37:21,529: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-18 02:37:21,529: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 02:37:29,920: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 02:37:30,223: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-18 02:37:30,224: INFO: main: *******************]
[2023-11-18 02:37:30,224: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-18 02:37:30,225: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:37:30,226: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:37:30,226: INFO: common: created directory: artifacts]
[2023-11-18 02:37:30,227: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-18 02:37:30,505: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-18 02:37:30,618: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-18 02:37:30,618: INFO: main: *******************]
[2023-11-18 02:37:30,618: INFO: main: >>>> stage: Training started <<<<]
[2023-11-18 02:37:30,619: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:37:30,621: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:37:30,621: INFO: common: created directory: artifacts]
[2023-11-18 02:37:30,621: INFO: common: created directory: artifacts/training]
[2023-11-18 02:37:30,622: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:37:44,204: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-18 02:37:44,205: INFO: main: *******************]
[2023-11-18 02:37:44,205: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-18 02:37:44,206: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:37:44,207: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:37:44,208: INFO: common: created directory: artifacts]
[2023-11-18 02:37:48,748: INFO: common: json file saved at scores.json]
[2023-11-18 02:37:48,748: INFO: common: json file saved at scores.json]
[2023-11-18 02:37:49,466: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-18 02:37:49,753: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmpls3we4wy\model\data\model\assets]
[2023-11-18 02:37:52,642: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\a99913c6ef204740b779258bd56c4666\artifacts\model\data]
[2023-11-18 02:37:52,643: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\a99913c6ef204740b779258bd56c4666\artifacts\model\data\model]
[2023-11-18 02:37:52,643: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\a99913c6ef204740b779258bd56c4666\artifacts\model\data\model\assets]
[2023-11-18 02:37:52,645: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\a99913c6ef204740b779258bd56c4666\artifacts\model\data\model\variables]
[2023-11-18 02:37:52,686: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-18 02:45:01,438: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-18 02:45:01,440: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:45:01,440: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:45:01,442: INFO: common: created directory: artifacts]
[2023-11-18 02:45:01,442: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-18 02:45:01,442: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 02:45:08,708: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 02:45:08,987: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-18 02:45:08,988: INFO: main: *******************]
[2023-11-18 02:45:08,988: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-18 02:45:08,989: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:45:08,990: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:45:08,990: INFO: common: created directory: artifacts]
[2023-11-18 02:45:08,991: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-18 02:45:09,228: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-18 02:45:09,356: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-18 02:45:09,356: INFO: main: *******************]
[2023-11-18 02:45:09,356: INFO: main: >>>> stage: Training started <<<<]
[2023-11-18 02:45:09,358: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:45:09,359: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:45:09,359: INFO: common: created directory: artifacts]
[2023-11-18 02:45:09,360: INFO: common: created directory: artifacts/training]
[2023-11-18 02:45:09,361: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:45:22,985: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-18 02:45:22,986: INFO: main: *******************]
[2023-11-18 02:45:22,986: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-18 02:45:22,987: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 02:45:22,988: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 02:45:22,989: INFO: common: created directory: artifacts]
[2023-11-18 02:45:27,451: INFO: common: json file saved at scores.json]
[2023-11-18 02:45:27,451: INFO: common: json file saved at scores.json]
[2023-11-18 02:45:42,657: ERROR: main: API request to endpoint /api/2.0/mlflow/runs/log-batch failed with error code 400 != 200. Response body: '"repo not associated with run"']
Traceback (most recent call last):
  File "main.py", line 47, in <module>
    model_evaluation.main()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\pipeline\stage_04_model_evaluation_mlflow.py", line 17, in main
    evaluation.log_into_mlflow()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_evaluation_mlflow.py", line 57, in log_into_mlflow
    mlflow.log_params(self.config.all_params)
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\mlflow\tracking\fluent.py", line 702, in log_params
    MlflowClient().log_batch(run_id=run_id, metrics=[], params=params_arr, tags=[])
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\mlflow\tracking\client.py", line 986, in log_batch
    self._tracking_client.log_batch(run_id, metrics, params, tags)
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\mlflow\tracking\_tracking_service\client.py", line 389, in log_batch
    self.store.log_batch(
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\mlflow\store\tracking\rest_store.py", line 321, in log_batch
    self._call_endpoint(LogBatch, req_body)
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\mlflow\store\tracking\rest_store.py", line 56, in _call_endpoint
    return call_endpoint(self.get_host_creds(), endpoint, method, json_body, response_proto)
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\mlflow\utils\rest_utils.py", line 281, in call_endpoint
    response = verify_rest_response(response, endpoint)
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\mlflow\utils\rest_utils.py", line 213, in verify_rest_response
    raise MlflowException(
mlflow.exceptions.MlflowException: API request to endpoint /api/2.0/mlflow/runs/log-batch failed with error code 400 != 200. Response body: '"repo not associated with run"'
[2023-11-18 03:29:11,453: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-18 03:29:11,455: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 03:29:11,456: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 03:29:11,456: INFO: common: created directory: artifacts]
[2023-11-18 03:29:11,457: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-18 03:29:11,457: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 03:29:17,745: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 03:29:18,034: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-18 03:29:18,034: INFO: main: *******************]
[2023-11-18 03:29:18,034: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-18 03:29:18,035: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 03:29:18,035: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 03:29:18,036: INFO: common: created directory: artifacts]
[2023-11-18 03:29:18,036: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-18 03:29:18,262: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-18 03:29:18,392: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-18 03:29:18,393: INFO: main: *******************]
[2023-11-18 03:29:18,393: INFO: main: >>>> stage: Training started <<<<]
[2023-11-18 03:29:18,394: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 03:29:18,395: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 03:29:18,396: INFO: common: created directory: artifacts]
[2023-11-18 03:29:18,396: INFO: common: created directory: artifacts/training]
[2023-11-18 03:29:18,397: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 03:29:31,923: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-18 03:29:31,924: INFO: main: *******************]
[2023-11-18 03:29:31,924: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-18 03:29:31,925: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 03:29:31,926: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 03:29:31,926: INFO: common: created directory: artifacts]
[2023-11-18 03:29:36,302: INFO: common: json file saved at scores.json]
[2023-11-18 03:29:50,021: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-18 03:29:50,306: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmpz7g1xh4d\model\data\model\assets]
[2023-11-18 03:31:10,434: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-18 13:30:22,458: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-18 13:30:22,460: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 13:30:22,461: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 13:30:22,462: INFO: common: created directory: artifacts]
[2023-11-18 13:30:22,462: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-18 13:30:22,462: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 13:30:30,170: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 13:30:30,485: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-18 13:30:35,008: INFO: stage_02_prepare_base_model: *******************]
[2023-11-18 13:30:35,008: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage started <<<<]
[2023-11-18 13:30:35,010: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 13:30:35,011: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 13:30:35,011: INFO: common: created directory: artifacts]
[2023-11-18 13:30:35,011: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-18 13:30:35,239: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-18 13:30:35,353: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-18 13:30:37,459: INFO: stage_03_model_training: *******************]
[2023-11-18 13:30:37,459: INFO: stage_03_model_training: >>>> stage: Training started <<<<]
[2023-11-18 13:30:37,461: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 13:30:37,462: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 13:30:37,462: INFO: common: created directory: artifacts]
[2023-11-18 13:30:37,462: INFO: common: created directory: artifacts/training]
[2023-11-18 13:30:37,463: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 13:30:50,781: INFO: stage_03_model_training: >>>> stage: Training ended <<<<

x==========x]
[2023-11-18 13:32:14,415: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-18 13:32:14,417: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 13:32:14,418: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 13:32:14,418: INFO: common: created directory: artifacts]
[2023-11-18 13:32:14,419: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-18 13:32:14,419: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 13:32:21,433: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-18 13:32:21,730: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-18 13:32:25,185: INFO: stage_02_prepare_base_model: *******************]
[2023-11-18 13:32:25,185: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage started <<<<]
[2023-11-18 13:32:25,187: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 13:32:25,188: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 13:32:25,188: INFO: common: created directory: artifacts]
[2023-11-18 13:32:25,188: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-18 13:32:25,430: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-18 13:32:25,547: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-18 13:32:27,633: INFO: stage_03_model_training: *******************]
[2023-11-18 13:32:27,633: INFO: stage_03_model_training: >>>> stage: Training started <<<<]
[2023-11-18 13:32:27,634: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 13:32:27,635: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 13:32:27,636: INFO: common: created directory: artifacts]
[2023-11-18 13:32:27,636: INFO: common: created directory: artifacts/training]
[2023-11-18 13:32:27,637: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 13:32:40,979: INFO: stage_03_model_training: >>>> stage: Training ended <<<<

x==========x]
[2023-11-18 13:33:33,136: INFO: stage_04_model_evaluation_mlflow: *******************]
[2023-11-18 13:33:33,136: INFO: stage_04_model_evaluation_mlflow: >>>>>> stage Evaluation started <<<<<<]
[2023-11-18 13:33:33,138: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-18 13:33:33,139: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-18 13:33:33,139: INFO: common: created directory: artifacts]
[2023-11-18 13:33:37,535: INFO: common: json file saved at scores.json]
[2023-11-18 13:33:43,707: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-18 13:33:43,999: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmpfoq81hhj\model\data\model\assets]
[2023-11-18 13:35:36,094: INFO: stage_04_model_evaluation_mlflow: >>>>>> stage Evaluation completed <<<<<<

x==========x]
[2023-11-19 03:59:58,254: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 03:59:58,257: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 03:59:58,258: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 03:59:58,258: INFO: common: created directory: artifacts]
[2023-11-19 03:59:58,259: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 03:59:58,259: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 04:00:06,113: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 04:00:06,622: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-19 04:00:06,622: INFO: main: *******************]
[2023-11-19 04:00:06,623: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-19 04:00:06,625: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 04:00:06,627: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 04:00:06,628: INFO: common: created directory: artifacts]
[2023-11-19 04:00:06,628: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-19 04:00:07,121: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-19 04:00:07,613: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-19 04:00:07,613: INFO: main: *******************]
[2023-11-19 04:00:07,613: INFO: main: >>>> stage: Training started <<<<]
[2023-11-19 04:00:07,616: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 04:00:07,618: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 04:00:07,618: INFO: common: created directory: artifacts]
[2023-11-19 04:00:07,618: INFO: common: created directory: artifacts/training]
[2023-11-19 04:00:07,621: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 04:00:51,402: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-19 04:00:51,403: INFO: main: *******************]
[2023-11-19 04:00:51,403: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-19 04:00:51,404: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 04:00:51,405: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 04:00:51,405: INFO: common: created directory: artifacts]
[2023-11-19 04:00:56,091: INFO: common: json file saved at scores.json]
[2023-11-19 04:00:56,881: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-19 04:00:57,200: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmp79u7qnja\model\data\model\assets]
[2023-11-19 04:01:00,872: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\fd53260e80404cdd9539793e62b3b97e\artifacts\model\data]
[2023-11-19 04:01:00,873: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\fd53260e80404cdd9539793e62b3b97e\artifacts\model\data\model]
[2023-11-19 04:01:00,874: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\fd53260e80404cdd9539793e62b3b97e\artifacts\model\data\model\assets]
[2023-11-19 04:01:00,875: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\fd53260e80404cdd9539793e62b3b97e\artifacts\model\data\model\variables]
[2023-11-19 04:01:00,923: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-19 04:53:19,901: ERROR: utils: Form data requires "python-multipart" to be installed. 
You can install "python-multipart" with: 

pip install python-multipart
]
[2023-11-19 06:48:09,831: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 06:48:09,833: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 06:48:09,834: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:48:09,835: INFO: common: created directory: artifacts]
[2023-11-19 06:48:09,835: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 06:48:09,835: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 06:48:15,263: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 06:48:15,566: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-19 06:48:15,566: INFO: main: *******************]
[2023-11-19 06:48:15,566: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-19 06:48:15,567: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 06:48:15,568: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:48:15,569: INFO: common: created directory: artifacts]
[2023-11-19 06:48:15,569: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-19 06:48:15,833: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-19 06:48:15,957: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-19 06:48:15,958: INFO: main: *******************]
[2023-11-19 06:48:15,958: INFO: main: >>>> stage: Training started <<<<]
[2023-11-19 06:48:15,959: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 06:48:15,960: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:48:15,960: INFO: common: created directory: artifacts]
[2023-11-19 06:48:15,960: INFO: common: created directory: artifacts/training]
[2023-11-19 06:48:15,961: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:48:29,651: ERROR: main: 'exist_ok' is an invalid keyword argument for mkdir()]
Traceback (most recent call last):
  File "main.py", line 35, in <module>
    model_trainer.main()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\pipeline\stage_03_model_training.py", line 17, in main
    training.train()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 84, in train
    os.mkdir('model/', exist_ok=True)
TypeError: 'exist_ok' is an invalid keyword argument for mkdir()
[2023-11-19 06:50:46,404: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 06:50:46,405: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 06:50:46,406: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:50:46,407: INFO: common: created directory: artifacts]
[2023-11-19 06:50:46,407: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 06:50:46,407: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 06:50:51,501: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 06:50:51,772: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-19 06:50:51,773: INFO: main: *******************]
[2023-11-19 06:50:51,773: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-19 06:50:51,774: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 06:50:51,776: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:50:51,776: INFO: common: created directory: artifacts]
[2023-11-19 06:50:51,776: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-19 06:50:52,017: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-19 06:50:52,143: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-19 06:50:52,143: INFO: main: *******************]
[2023-11-19 06:50:52,143: INFO: main: >>>> stage: Training started <<<<]
[2023-11-19 06:50:52,145: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 06:50:52,146: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:50:52,146: INFO: common: created directory: artifacts]
[2023-11-19 06:50:52,146: INFO: common: created directory: artifacts/training]
[2023-11-19 06:50:52,147: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:51:05,927: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-19 06:51:05,928: INFO: main: *******************]
[2023-11-19 06:51:05,928: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-19 06:51:05,929: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 06:51:05,930: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:51:05,930: INFO: common: created directory: artifacts]
[2023-11-19 06:51:10,551: INFO: common: json file saved at scores.json]
[2023-11-19 06:51:11,303: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-19 06:51:11,602: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmp6jv8lavo\model\data\model\assets]
[2023-11-19 06:51:15,109: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\2c7c8827015f4fcfb8d7495572303222\artifacts\model\data]
[2023-11-19 06:51:15,110: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\2c7c8827015f4fcfb8d7495572303222\artifacts\model\data\model]
[2023-11-19 06:51:15,110: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\2c7c8827015f4fcfb8d7495572303222\artifacts\model\data\model\assets]
[2023-11-19 06:51:15,112: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\2c7c8827015f4fcfb8d7495572303222\artifacts\model\data\model\variables]
[2023-11-19 06:51:15,156: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-19 06:53:15,598: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 06:53:15,600: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 06:53:15,601: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:53:15,601: INFO: common: created directory: artifacts]
[2023-11-19 06:53:15,601: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 06:53:15,602: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 06:53:20,847: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 06:53:21,144: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-19 06:53:24,659: INFO: stage_02_prepare_base_model: *******************]
[2023-11-19 06:53:24,660: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage started <<<<]
[2023-11-19 06:53:24,661: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 06:53:24,662: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:53:24,663: INFO: common: created directory: artifacts]
[2023-11-19 06:53:24,663: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-19 06:53:24,900: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-19 06:53:25,015: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-19 06:53:27,158: INFO: stage_03_model_training: *******************]
[2023-11-19 06:53:27,159: INFO: stage_03_model_training: >>>> stage: Training started <<<<]
[2023-11-19 06:53:27,160: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 06:53:27,161: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:53:27,162: INFO: common: created directory: artifacts]
[2023-11-19 06:53:27,162: INFO: common: created directory: artifacts/training]
[2023-11-19 06:53:27,163: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:53:40,670: INFO: stage_03_model_training: >>>> stage: Training ended <<<<

x==========x]
[2023-11-19 06:53:43,213: INFO: stage_04_model_evaluation_mlflow: *******************]
[2023-11-19 06:53:43,214: INFO: stage_04_model_evaluation_mlflow: >>>>>> stage Evaluation started <<<<<<]
[2023-11-19 06:53:43,215: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 06:53:43,216: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 06:53:43,216: INFO: common: created directory: artifacts]
[2023-11-19 06:53:47,830: INFO: common: json file saved at scores.json]
[2023-11-19 06:53:48,489: WARNING: save: Found untraced functions such as _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op, _jit_compiled_convolution_op while saving (showing 5 of 14). These functions will not be directly callable after loading.]
[2023-11-19 06:53:48,814: INFO: builder_impl: Assets written to: C:\Users\MASTE\AppData\Local\Temp\tmpts0os7i0\model\data\model\assets]
[2023-11-19 06:53:52,072: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\001eadbdd3f241fcaf4bb13ccd786b09\artifacts\model\data]
[2023-11-19 06:53:52,073: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\001eadbdd3f241fcaf4bb13ccd786b09\artifacts\model\data\model]
[2023-11-19 06:53:52,074: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\001eadbdd3f241fcaf4bb13ccd786b09\artifacts\model\data\model\assets]
[2023-11-19 06:53:52,075: INFO: dir_util: creating D:\My DL Workstation\Krish\Complete End to End Deep Learning Project\mlruns\0\001eadbdd3f241fcaf4bb13ccd786b09\artifacts\model\data\model\variables]
[2023-11-19 06:53:52,117: INFO: stage_04_model_evaluation_mlflow: >>>>>> stage Evaluation completed <<<<<<

x==========x]
[2023-11-19 17:19:26,780: INFO: stage_02_prepare_base_model: *******************]
[2023-11-19 17:19:26,780: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage started <<<<]
[2023-11-19 17:19:26,781: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 17:19:26,783: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 17:19:26,783: INFO: common: created directory: artifacts]
[2023-11-19 17:19:26,783: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-19 17:19:27,010: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-19 17:19:27,123: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-19 17:19:29,406: INFO: stage_03_model_training: *******************]
[2023-11-19 17:19:29,406: INFO: stage_03_model_training: >>>> stage: Training started <<<<]
[2023-11-19 17:19:29,408: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 17:19:29,409: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 17:19:29,409: INFO: common: created directory: artifacts]
[2023-11-19 17:19:29,410: INFO: common: created directory: artifacts/training]
[2023-11-19 17:19:29,411: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 17:19:30,852: ERROR: stage_03_model_training: Graph execution error:

Detected at node 'categorical_crossentropy/softmax_cross_entropy_with_logits' defined at (most recent call last):
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 24, in <module>
      obj.main()
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 17, in main
      training.train()
    File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 70, in train
      self.model.fit(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1685, in fit
      tmp_logs = self.train_function(iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1284, in train_function
      return step_function(self, iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1268, in step_function
      outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1249, in run_step
      outputs = model.train_step(data)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1051, in train_step
      loss = self.compute_loss(x, y, y_pred, sample_weight)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1109, in compute_loss
      return self.compiled_loss(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\compile_utils.py", line 265, in __call__
      loss_value = loss_obj(y_t, y_p, sample_weight=sw)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 142, in __call__
      losses = call_fn(y_true, y_pred)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 268, in call
      return ag_fn(y_true, y_pred, **self._fn_kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 1984, in categorical_crossentropy
      return backend.categorical_crossentropy(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\backend.py", line 5565, in categorical_crossentropy
      return tf.nn.softmax_cross_entropy_with_logits(
Node: 'categorical_crossentropy/softmax_cross_entropy_with_logits'
logits and labels must be broadcastable: logits_size=[16,4] labels_size=[16,2]
	 [[{{node categorical_crossentropy/softmax_cross_entropy_with_logits}}]] [Op:__inference_train_function_881]]
Traceback (most recent call last):
  File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 24, in <module>
    obj.main()
  File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 17, in main
    training.train()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 70, in train
    self.model.fit(
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 70, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\tensorflow\python\eager\execute.py", line 52, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.InvalidArgumentError: Graph execution error:

Detected at node 'categorical_crossentropy/softmax_cross_entropy_with_logits' defined at (most recent call last):
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 24, in <module>
      obj.main()
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 17, in main
      training.train()
    File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 70, in train
      self.model.fit(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1685, in fit
      tmp_logs = self.train_function(iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1284, in train_function
      return step_function(self, iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1268, in step_function
      outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1249, in run_step
      outputs = model.train_step(data)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1051, in train_step
      loss = self.compute_loss(x, y, y_pred, sample_weight)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1109, in compute_loss
      return self.compiled_loss(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\compile_utils.py", line 265, in __call__
      loss_value = loss_obj(y_t, y_p, sample_weight=sw)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 142, in __call__
      losses = call_fn(y_true, y_pred)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 268, in call
      return ag_fn(y_true, y_pred, **self._fn_kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 1984, in categorical_crossentropy
      return backend.categorical_crossentropy(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\backend.py", line 5565, in categorical_crossentropy
      return tf.nn.softmax_cross_entropy_with_logits(
Node: 'categorical_crossentropy/softmax_cross_entropy_with_logits'
logits and labels must be broadcastable: logits_size=[16,4] labels_size=[16,2]
	 [[{{node categorical_crossentropy/softmax_cross_entropy_with_logits}}]] [Op:__inference_train_function_881]
[2023-11-19 17:22:08,363: INFO: stage_03_model_training: *******************]
[2023-11-19 17:22:08,364: INFO: stage_03_model_training: >>>> stage: Training started <<<<]
[2023-11-19 17:22:08,365: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 17:22:08,366: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 17:22:08,367: INFO: common: created directory: artifacts]
[2023-11-19 17:22:08,368: INFO: common: created directory: artifacts/training]
[2023-11-19 17:22:08,369: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 17:22:09,750: ERROR: stage_03_model_training: Graph execution error:

Detected at node 'categorical_crossentropy/softmax_cross_entropy_with_logits' defined at (most recent call last):
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 24, in <module>
      obj.main()
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 17, in main
      training.train()
    File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 70, in train
      self.model.fit(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1685, in fit
      tmp_logs = self.train_function(iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1284, in train_function
      return step_function(self, iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1268, in step_function
      outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1249, in run_step
      outputs = model.train_step(data)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1051, in train_step
      loss = self.compute_loss(x, y, y_pred, sample_weight)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1109, in compute_loss
      return self.compiled_loss(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\compile_utils.py", line 265, in __call__
      loss_value = loss_obj(y_t, y_p, sample_weight=sw)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 142, in __call__
      losses = call_fn(y_true, y_pred)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 268, in call
      return ag_fn(y_true, y_pred, **self._fn_kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 1984, in categorical_crossentropy
      return backend.categorical_crossentropy(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\backend.py", line 5565, in categorical_crossentropy
      return tf.nn.softmax_cross_entropy_with_logits(
Node: 'categorical_crossentropy/softmax_cross_entropy_with_logits'
logits and labels must be broadcastable: logits_size=[16,4] labels_size=[16,2]
	 [[{{node categorical_crossentropy/softmax_cross_entropy_with_logits}}]] [Op:__inference_train_function_881]]
Traceback (most recent call last):
  File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 24, in <module>
    obj.main()
  File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 17, in main
    training.train()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 70, in train
    self.model.fit(
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 70, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\tensorflow\python\eager\execute.py", line 52, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.InvalidArgumentError: Graph execution error:

Detected at node 'categorical_crossentropy/softmax_cross_entropy_with_logits' defined at (most recent call last):
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 24, in <module>
      obj.main()
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 17, in main
      training.train()
    File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 70, in train
      self.model.fit(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1685, in fit
      tmp_logs = self.train_function(iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1284, in train_function
      return step_function(self, iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1268, in step_function
      outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1249, in run_step
      outputs = model.train_step(data)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1051, in train_step
      loss = self.compute_loss(x, y, y_pred, sample_weight)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1109, in compute_loss
      return self.compiled_loss(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\compile_utils.py", line 265, in __call__
      loss_value = loss_obj(y_t, y_p, sample_weight=sw)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 142, in __call__
      losses = call_fn(y_true, y_pred)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 268, in call
      return ag_fn(y_true, y_pred, **self._fn_kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 1984, in categorical_crossentropy
      return backend.categorical_crossentropy(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\backend.py", line 5565, in categorical_crossentropy
      return tf.nn.softmax_cross_entropy_with_logits(
Node: 'categorical_crossentropy/softmax_cross_entropy_with_logits'
logits and labels must be broadcastable: logits_size=[16,4] labels_size=[16,2]
	 [[{{node categorical_crossentropy/softmax_cross_entropy_with_logits}}]] [Op:__inference_train_function_881]
[2023-11-19 19:03:56,867: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 19:03:56,869: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 19:03:56,870: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:03:56,870: INFO: common: created directory: artifacts]
[2023-11-19 19:03:56,870: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 19:03:56,871: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 19:04:02,513: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 19:04:02,513: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-19 19:04:35,918: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 19:04:35,920: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 19:04:35,920: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:04:35,921: INFO: common: created directory: artifacts]
[2023-11-19 19:04:35,921: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 19:04:35,922: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 19:04:41,110: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 19:04:41,111: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-19 19:05:35,046: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 19:05:35,047: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 19:05:35,048: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:05:35,049: INFO: common: created directory: artifacts]
[2023-11-19 19:05:35,049: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 19:05:35,049: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 19:05:39,928: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-19 19:05:39,928: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-19 19:23:51,620: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 19:23:51,622: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 19:23:51,623: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:23:51,623: INFO: common: created directory: artifacts]
[2023-11-19 19:23:51,623: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 19:23:51,624: INFO: data_ingestion: Data download and extraction skipped due to config.yaml skip_down being enabled.]
[2023-11-19 19:23:51,624: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-19 19:24:56,171: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 19:24:56,177: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 19:24:56,178: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:24:56,179: INFO: common: created directory: artifacts]
[2023-11-19 19:24:56,179: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 19:24:56,179: INFO: data_ingestion: Data download and extraction skipped due to config.yaml skip_down being enabled.]
[2023-11-19 19:24:56,180: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-19 19:32:46,477: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 19:32:46,478: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 19:32:46,479: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:32:46,480: INFO: common: created directory: artifacts]
[2023-11-19 19:32:46,480: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 19:32:46,480: INFO: data_ingestion: Data download skipped due to config.yaml skip_down being enabled.]
[2023-11-19 19:32:46,480: INFO: data_ingestion: Data extraction skipped due to config.yaml skip_down being enabled.]
[2023-11-19 19:32:46,480: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-19 19:32:46,481: INFO: main: *******************]
[2023-11-19 19:32:46,481: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-19 19:32:46,482: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 19:32:46,483: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:32:46,483: INFO: common: created directory: artifacts]
[2023-11-19 19:32:46,484: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-19 19:32:46,707: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-19 19:32:46,823: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-19 19:32:46,823: INFO: main: *******************]
[2023-11-19 19:32:46,823: INFO: main: >>>> stage: Training started <<<<]
[2023-11-19 19:32:46,825: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 19:32:46,826: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:32:46,826: INFO: common: created directory: artifacts]
[2023-11-19 19:32:46,826: INFO: common: created directory: artifacts/training]
[2023-11-19 19:32:46,827: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:32:47,003: ERROR: main: [WinError 3] The system cannot find the path specified: 'artifacts\\data_ingestion\\kidney-ct-scan-image']
Traceback (most recent call last):
  File "main.py", line 35, in <module>
    model_trainer.main()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\pipeline\stage_03_model_training.py", line 16, in main
    training.train_val_generator()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 32, in train_val_generator
    self.valid_generator = valid_datagenerator.flow_from_directory(
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\preprocessing\image.py", line 1648, in flow_from_directory
    return DirectoryIterator(
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\preprocessing\image.py", line 563, in __init__
    for subdir in sorted(os.listdir(directory)):
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'artifacts\\data_ingestion\\kidney-ct-scan-image'
[2023-11-19 19:33:17,348: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 19:33:17,350: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 19:33:17,351: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:33:17,351: INFO: common: created directory: artifacts]
[2023-11-19 19:33:17,352: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 19:33:17,352: INFO: data_ingestion: Data download skipped due to config.yaml skip_down being enabled.]
[2023-11-19 19:33:17,352: INFO: data_ingestion: Data extraction skipped due to config.yaml skip_down being enabled.]
[2023-11-19 19:33:17,352: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-19 19:33:17,352: INFO: main: *******************]
[2023-11-19 19:33:17,352: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-19 19:33:17,353: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 19:33:17,354: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:33:17,355: INFO: common: created directory: artifacts]
[2023-11-19 19:33:17,355: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-19 19:33:17,581: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-19 19:33:17,708: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-19 19:33:17,709: INFO: main: *******************]
[2023-11-19 19:33:17,709: INFO: main: >>>> stage: Training started <<<<]
[2023-11-19 19:33:17,710: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 19:33:17,711: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 19:33:17,711: INFO: common: created directory: artifacts]
[2023-11-19 19:33:17,711: INFO: common: created directory: artifacts/training]
[2023-11-19 19:33:17,712: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 23:42:53,723: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 23:42:53,725: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 23:42:53,726: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 23:42:53,726: INFO: common: created directory: artifacts]
[2023-11-19 23:42:53,727: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 23:42:53,727: INFO: data_ingestion: Data download skipped due to config.yaml skip_down being enabled.]
[2023-11-19 23:42:53,727: INFO: data_ingestion: Data extraction skipped due to config.yaml skip_down being enabled.]
[2023-11-19 23:42:53,727: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-19 23:45:05,600: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 23:45:05,602: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 23:45:05,603: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 23:45:05,603: INFO: common: created directory: artifacts]
[2023-11-19 23:45:05,603: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 23:45:05,603: INFO: data_ingestion: Data download skipped due to config.yaml skip_down being enabled.]
[2023-11-19 23:45:05,603: INFO: data_ingestion: Data extraction skipped due to config.yaml skip_down being enabled.]
[2023-11-19 23:45:05,604: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-19 23:45:05,604: INFO: main: *******************]
[2023-11-19 23:45:05,604: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-19 23:45:05,605: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 23:45:05,606: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 23:45:05,607: INFO: common: created directory: artifacts]
[2023-11-19 23:45:05,607: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-19 23:45:06,638: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-19 23:45:06,787: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-19 23:45:06,788: INFO: main: *******************]
[2023-11-19 23:45:06,788: INFO: main: >>>> stage: Training started <<<<]
[2023-11-19 23:45:06,789: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 23:45:06,790: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 23:45:06,791: INFO: common: created directory: artifacts]
[2023-11-19 23:45:06,791: INFO: common: created directory: artifacts/training]
[2023-11-19 23:45:06,792: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 23:45:06,990: ERROR: main: Asked to retrieve element 0, but the Sequence has length 0]
Traceback (most recent call last):
  File "main.py", line 35, in <module>
    model_trainer.main()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\pipeline\stage_03_model_training.py", line 17, in main
    training.train()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 69, in train
    self.model.fit(
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 70, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\preprocessing\image.py", line 103, in __getitem__
    raise ValueError(
ValueError: Asked to retrieve element 0, but the Sequence has length 0
[2023-11-19 23:46:32,701: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-19 23:46:32,702: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 23:46:32,703: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 23:46:32,703: INFO: common: created directory: artifacts]
[2023-11-19 23:46:32,704: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-19 23:46:32,704: INFO: data_ingestion: Data download skipped due to config.yaml skip_down being enabled.]
[2023-11-19 23:46:32,704: INFO: data_ingestion: Data extraction skipped due to config.yaml skip_down being enabled.]
[2023-11-19 23:46:32,704: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-19 23:46:32,704: INFO: main: *******************]
[2023-11-19 23:46:32,705: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-19 23:46:32,706: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 23:46:32,707: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 23:46:32,707: INFO: common: created directory: artifacts]
[2023-11-19 23:46:32,707: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-19 23:46:33,584: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-19 23:46:33,741: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-19 23:46:33,741: INFO: main: *******************]
[2023-11-19 23:46:33,741: INFO: main: >>>> stage: Training started <<<<]
[2023-11-19 23:46:33,743: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 23:46:33,743: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 23:46:33,743: INFO: common: created directory: artifacts]
[2023-11-19 23:46:33,743: INFO: common: created directory: artifacts/training]
[2023-11-19 23:46:33,744: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 23:59:35,946: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-19 23:59:35,947: INFO: main: *******************]
[2023-11-19 23:59:35,947: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-19 23:59:35,948: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-19 23:59:35,949: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-19 23:59:35,949: INFO: common: created directory: artifacts]
[2023-11-19 23:59:48,621: INFO: common: json file saved at scores.json]
[2023-11-19 23:59:48,621: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-20 00:15:55,841: INFO: main: >>>> stage: Data ingestion stage started <<<<]
[2023-11-20 00:15:55,843: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 00:15:55,844: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 00:15:55,844: INFO: common: created directory: artifacts]
[2023-11-20 00:15:55,844: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-20 00:15:55,844: INFO: data_ingestion: Data download skipped due to config.yaml skip_down being enabled.]
[2023-11-20 00:15:55,844: INFO: data_ingestion: Data extraction skipped due to config.yaml skip_down being enabled.]
[2023-11-20 00:15:55,845: INFO: main: >>>> stage: Data ingestion stage ended <<<<

x==========x]
[2023-11-20 00:15:55,845: INFO: main: *******************]
[2023-11-20 00:15:55,845: INFO: main: >>>> stage: Prepare base model stage started <<<<]
[2023-11-20 00:15:55,847: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 00:15:55,848: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 00:15:55,848: INFO: common: created directory: artifacts]
[2023-11-20 00:15:55,848: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-20 00:15:56,746: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-20 00:15:56,975: INFO: main: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-20 00:15:56,976: INFO: main: *******************]
[2023-11-20 00:15:56,976: INFO: main: >>>> stage: Training started <<<<]
[2023-11-20 00:15:56,978: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 00:15:56,979: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 00:15:56,979: INFO: common: created directory: artifacts]
[2023-11-20 00:15:56,979: INFO: common: created directory: artifacts/training]
[2023-11-20 00:15:56,980: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 02:41:36,301: INFO: main: >>>> stage: Training ended <<<<

x==========x]
[2023-11-20 02:41:36,301: INFO: main: *******************]
[2023-11-20 02:41:36,302: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-20 02:41:36,304: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 02:41:36,305: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 02:41:36,305: INFO: common: created directory: artifacts]
[2023-11-20 02:41:48,585: INFO: common: json file saved at scores.json]
[2023-11-20 02:41:48,585: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-20 02:42:51,686: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-20 02:42:51,688: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 02:42:51,689: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 02:42:51,690: INFO: common: created directory: artifacts]
[2023-11-20 02:42:51,690: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-20 02:42:51,690: INFO: data_ingestion: Data download skipped due to config.yaml skip_down being enabled.]
[2023-11-20 02:42:51,690: INFO: data_ingestion: Data extraction skipped due to config.yaml skip_down being enabled.]
[2023-11-20 02:42:51,691: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-20 03:07:18,348: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-20 03:07:18,350: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 03:07:18,351: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:07:18,352: INFO: common: created directory: artifacts]
[2023-11-20 03:07:18,352: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-20 03:07:18,352: INFO: data_ingestion: Data download skipped due to config.yaml skip_down being enabled.]
[2023-11-20 03:07:18,352: INFO: data_ingestion: Data extraction skipped due to config.yaml skip_down being enabled.]
[2023-11-20 03:07:18,353: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-20 03:07:50,923: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage started <<<<]
[2023-11-20 03:07:50,925: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 03:07:50,926: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:07:50,926: INFO: common: created directory: artifacts]
[2023-11-20 03:07:50,926: INFO: common: created directory: artifacts/data_ingestion]
[2023-11-20 03:07:50,927: INFO: data_ingestion: Downloading data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-20 03:07:58,476: INFO: data_ingestion: Downloaded data from: https://drive.google.com/file/d/1vlhZ5c7abUKF8xXERIw6m9Te8fW7ohw3/view?usp=sharing into file: artifacts/data_ingestion/data.zip]
[2023-11-20 03:07:58,786: INFO: data_ingestion: Downloaded data extracted from: artifacts/data_ingestion into: artifacts/data_ingestion]
[2023-11-20 03:07:58,787: INFO: stage_01_data_ingestion: >>>> stage: Data ingestion stage ended <<<<]
[2023-11-20 03:08:24,854: INFO: stage_02_prepare_base_model: *******************]
[2023-11-20 03:08:24,854: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage started <<<<]
[2023-11-20 03:08:24,855: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 03:08:24,856: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:08:24,857: INFO: common: created directory: artifacts]
[2023-11-20 03:08:24,857: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-20 03:08:26,382: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-20 03:08:26,748: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-20 03:08:28,826: INFO: stage_03_model_training: *******************]
[2023-11-20 03:08:28,827: INFO: stage_03_model_training: >>>> stage: Training started <<<<]
[2023-11-20 03:08:28,828: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 03:08:28,829: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:08:28,830: INFO: common: created directory: artifacts]
[2023-11-20 03:08:28,830: INFO: common: created directory: artifacts/training]
[2023-11-20 03:08:28,831: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:08:35,990: ERROR: stage_03_model_training: Graph execution error:

Detected at node 'categorical_crossentropy/softmax_cross_entropy_with_logits' defined at (most recent call last):
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 24, in <module>
      obj.main()
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 17, in main
      training.train()
    File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 69, in train
      self.model.fit(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1564, in fit
      tmp_logs = self.train_function(iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1160, in train_function
      return step_function(self, iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1146, in step_function
      outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1135, in run_step
      outputs = model.train_step(data)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 994, in train_step
      loss = self.compute_loss(x, y, y_pred, sample_weight)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1052, in compute_loss
      return self.compiled_loss(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\compile_utils.py", line 265, in __call__
      loss_value = loss_obj(y_t, y_p, sample_weight=sw)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 152, in __call__
      losses = call_fn(y_true, y_pred)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 272, in call
      return ag_fn(y_true, y_pred, **self._fn_kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 1990, in categorical_crossentropy
      return backend.categorical_crossentropy(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\backend.py", line 5535, in categorical_crossentropy
      return tf.nn.softmax_cross_entropy_with_logits(
Node: 'categorical_crossentropy/softmax_cross_entropy_with_logits'
logits and labels must be broadcastable: logits_size=[16,4] labels_size=[16,2]
	 [[{{node categorical_crossentropy/softmax_cross_entropy_with_logits}}]] [Op:__inference_train_function_1201]]
Traceback (most recent call last):
  File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 24, in <module>
    obj.main()
  File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 17, in main
    training.train()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 69, in train
    self.model.fit(
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 70, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\tensorflow\python\eager\execute.py", line 54, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.InvalidArgumentError: Graph execution error:

Detected at node 'categorical_crossentropy/softmax_cross_entropy_with_logits' defined at (most recent call last):
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 24, in <module>
      obj.main()
    File "src\CNNClassifier\pipeline\stage_03_model_training.py", line 17, in main
      training.train()
    File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_training.py", line 69, in train
      self.model.fit(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1564, in fit
      tmp_logs = self.train_function(iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1160, in train_function
      return step_function(self, iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1146, in step_function
      outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1135, in run_step
      outputs = model.train_step(data)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 994, in train_step
      loss = self.compute_loss(x, y, y_pred, sample_weight)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1052, in compute_loss
      return self.compiled_loss(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\compile_utils.py", line 265, in __call__
      loss_value = loss_obj(y_t, y_p, sample_weight=sw)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 152, in __call__
      losses = call_fn(y_true, y_pred)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 272, in call
      return ag_fn(y_true, y_pred, **self._fn_kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 1990, in categorical_crossentropy
      return backend.categorical_crossentropy(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\backend.py", line 5535, in categorical_crossentropy
      return tf.nn.softmax_cross_entropy_with_logits(
Node: 'categorical_crossentropy/softmax_cross_entropy_with_logits'
logits and labels must be broadcastable: logits_size=[16,4] labels_size=[16,2]
	 [[{{node categorical_crossentropy/softmax_cross_entropy_with_logits}}]] [Op:__inference_train_function_1201]
[2023-11-20 03:08:57,142: INFO: stage_02_prepare_base_model: *******************]
[2023-11-20 03:08:57,142: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage started <<<<]
[2023-11-20 03:08:57,144: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 03:08:57,145: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:08:57,145: INFO: common: created directory: artifacts]
[2023-11-20 03:08:57,145: INFO: common: created directory: artifacts/prepare_base_model]
[2023-11-20 03:08:58,124: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2023-11-20 03:08:58,476: INFO: stage_02_prepare_base_model: >>>> stage: Prepare base model stage ended <<<<

x==========x]
[2023-11-20 03:09:00,532: INFO: stage_03_model_training: *******************]
[2023-11-20 03:09:00,532: INFO: stage_03_model_training: >>>> stage: Training started <<<<]
[2023-11-20 03:09:00,534: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 03:09:00,535: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:09:00,535: INFO: common: created directory: artifacts]
[2023-11-20 03:09:00,536: INFO: common: created directory: artifacts/training]
[2023-11-20 03:09:00,537: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:09:09,307: INFO: stage_03_model_training: >>>> stage: Training ended <<<<

x==========x]
[2023-11-20 03:09:13,623: INFO: stage_04_model_evaluation_mlflow: *******************]
[2023-11-20 03:09:13,623: INFO: stage_04_model_evaluation_mlflow: >>>>>> stage Evaluation started <<<<<<]
[2023-11-20 03:09:13,624: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 03:09:13,626: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:09:13,626: INFO: common: created directory: artifacts]
[2023-11-20 03:09:19,914: INFO: common: json file saved at scores.json]
[2023-11-20 03:09:19,915: INFO: stage_04_model_evaluation_mlflow: >>>>>> stage Evaluation completed <<<<<<

x==========x]
[2023-11-20 03:24:44,462: INFO: main: *******************]
[2023-11-20 03:24:44,462: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-20 03:24:44,464: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 03:24:44,466: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:24:44,466: INFO: common: created directory: artifacts]
[2023-11-20 03:24:51,110: INFO: common: json file saved at scores.json]
[2023-11-20 03:24:51,110: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2023-11-20 03:25:26,420: INFO: main: *******************]
[2023-11-20 03:25:26,420: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-20 03:25:26,421: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 03:25:26,423: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:25:26,423: INFO: common: created directory: artifacts]
[2023-11-20 03:25:32,018: ERROR: main: Graph execution error:

Detected at node 'categorical_crossentropy/softmax_cross_entropy_with_logits' defined at (most recent call last):
    File "main.py", line 47, in <module>
      model_evaluation.main()
    File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\pipeline\stage_04_model_evaluation_mlflow.py", line 15, in main
      evaluation.evaluation()
    File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_evaluation_mlflow.py", line 44, in evaluation
      self.score = self.model.evaluate(self.valid_generator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1947, in evaluate
      tmp_logs = self.test_function(iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1727, in test_function
      return step_function(self, iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1713, in step_function
      outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1701, in run_step
      outputs = model.test_step(data)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1667, in test_step
      self.compute_loss(x, y, y_pred, sample_weight)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1052, in compute_loss
      return self.compiled_loss(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\compile_utils.py", line 265, in __call__
      loss_value = loss_obj(y_t, y_p, sample_weight=sw)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 152, in __call__
      losses = call_fn(y_true, y_pred)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 272, in call
      return ag_fn(y_true, y_pred, **self._fn_kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 1990, in categorical_crossentropy
      return backend.categorical_crossentropy(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\backend.py", line 5535, in categorical_crossentropy
      return tf.nn.softmax_cross_entropy_with_logits(
Node: 'categorical_crossentropy/softmax_cross_entropy_with_logits'
logits and labels must be broadcastable: logits_size=[16,4] labels_size=[16,2]
	 [[{{node categorical_crossentropy/softmax_cross_entropy_with_logits}}]] [Op:__inference_test_function_723]]
Traceback (most recent call last):
  File "main.py", line 47, in <module>
    model_evaluation.main()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\pipeline\stage_04_model_evaluation_mlflow.py", line 15, in main
    evaluation.evaluation()
  File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_evaluation_mlflow.py", line 44, in evaluation
    self.score = self.model.evaluate(self.valid_generator)
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 70, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\tensorflow\python\eager\execute.py", line 54, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.InvalidArgumentError: Graph execution error:

Detected at node 'categorical_crossentropy/softmax_cross_entropy_with_logits' defined at (most recent call last):
    File "main.py", line 47, in <module>
      model_evaluation.main()
    File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\pipeline\stage_04_model_evaluation_mlflow.py", line 15, in main
      evaluation.evaluation()
    File "d:\my dl workstation\krish\complete end to end deep learning project\src\CNNClassifier\components\model_evaluation_mlflow.py", line 44, in evaluation
      self.score = self.model.evaluate(self.valid_generator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\utils\traceback_utils.py", line 65, in error_handler
      return fn(*args, **kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1947, in evaluate
      tmp_logs = self.test_function(iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1727, in test_function
      return step_function(self, iterator)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1713, in step_function
      outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1701, in run_step
      outputs = model.test_step(data)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1667, in test_step
      self.compute_loss(x, y, y_pred, sample_weight)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\training.py", line 1052, in compute_loss
      return self.compiled_loss(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\engine\compile_utils.py", line 265, in __call__
      loss_value = loss_obj(y_t, y_p, sample_weight=sw)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 152, in __call__
      losses = call_fn(y_true, y_pred)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 272, in call
      return ag_fn(y_true, y_pred, **self._fn_kwargs)
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\losses.py", line 1990, in categorical_crossentropy
      return backend.categorical_crossentropy(
    File "C:\Users\MASTE\anaconda3\envs\kidney\lib\site-packages\keras\backend.py", line 5535, in categorical_crossentropy
      return tf.nn.softmax_cross_entropy_with_logits(
Node: 'categorical_crossentropy/softmax_cross_entropy_with_logits'
logits and labels must be broadcastable: logits_size=[16,4] labels_size=[16,2]
	 [[{{node categorical_crossentropy/softmax_cross_entropy_with_logits}}]] [Op:__inference_test_function_723]
[2023-11-20 03:27:11,338: INFO: main: *******************]
[2023-11-20 03:27:11,338: INFO: main: >>>> stage: Evaluation started <<<<]
[2023-11-20 03:27:11,339: INFO: common: yaml file: config\config.yaml loaded successfully]
[2023-11-20 03:27:11,341: INFO: common: yaml file: params.yaml loaded successfully]
[2023-11-20 03:27:11,341: INFO: common: created directory: artifacts]
[2023-11-20 03:27:32,745: INFO: common: json file saved at scores.json]
[2023-11-20 03:27:32,745: INFO: main: >>>> stage: Evaluation ended <<<<

x==========x]
[2024-01-04 01:55:17,125: WARNING: def_function: 5 out of the last 5 calls to <function Model.make_predict_function.<locals>.predict_function at 0x0000017F59F43CA0> triggered tf.function retracing. Tracing is expensive and the excessive number of tracings could be due to (1) creating @tf.function repeatedly in a loop, (2) passing tensors with different shapes, (3) passing Python objects instead of tensors. For (1), please define your @tf.function outside of the loop. For (2), @tf.function has reduce_retracing=True option that can avoid unnecessary retracing. For (3), please refer to https://www.tensorflow.org/guide/function#controlling_retracing and https://www.tensorflow.org/api_docs/python/tf/function for  more details.]
[2024-01-04 01:55:32,003: WARNING: def_function: 6 out of the last 6 calls to <function Model.make_predict_function.<locals>.predict_function at 0x0000017F59F431F0> triggered tf.function retracing. Tracing is expensive and the excessive number of tracings could be due to (1) creating @tf.function repeatedly in a loop, (2) passing tensors with different shapes, (3) passing Python objects instead of tensors. For (1), please define your @tf.function outside of the loop. For (2), @tf.function has reduce_retracing=True option that can avoid unnecessary retracing. For (3), please refer to https://www.tensorflow.org/guide/function#controlling_retracing and https://www.tensorflow.org/api_docs/python/tf/function for  more details.]
